CMakeLists.txt

```diff
@@ -238,24 +235,59 @@ if (enable_checksums)
   set(SRC ${SRC} checksums.c)
 endif(enable_checksums)
 
-add_library(gc-lib STATIC ${SRC})
-set_target_properties(gc-lib PROPERTIES
-                      COMPILE_DEFINITIONS GC_NOT_DLL)
-#TODO TARGET_LINK_LIBRARIES(...  ...  ${LIBS})
-
-add_library(gcmt-lib STATIC ${SRC})
-set_target_properties(gcmt-lib PROPERTIES
-                      COMPILE_DEFINITIONS GC_NOT_DLL)
+# Create target
+add_library(gc ${SRC})
+target_include_directories(gc  
+    PUBLIC     
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
+)
+if (NOT BUILD_SHARED_LIBS)
+    target_compile_definitions(gc PRIVATE GC_NOT_DLL)
+else()
+    target_compile_definitions(gc PRIVATE GC_DLL)
+endif()
 
-add_library(gcmt-dll SHARED ${SRC})
+# Install library
+install(
+    TARGETS gc
+    LIBRARY DESTINATION lib 
```

Comment:
A feature to consider would be exporting the target, so that this CMakeLists generates `gc-config.cmake` and `gc-targets.cmake`, which would make linking to `gc` easier for downstream consumers. i.e.: it would allow `find_package(gc)` to work out-of-the-box.

Reply:
OK